rust/gitxetcore/src/summaries/analysis.rs

```diff
@@ -27,6 +29,9 @@ impl FileAnalyzers {
         if let Some(twb) = &mut self.twb {
             twb.process_chunk(chunk);
         }
+        if let Some(tds) = &mut self.tds {
+            tds.process_chunk(chunk);
+        }
```

Comment:
if you need to add one more ever, then maybe we make this be dynamic dispatch with an more OOP style.

```rust
pub trait FileAnalyzer {
  fn process_chunk(&self, chunk: &[u8]) -> Result<()>
}

pub struct FileAnalyers {
  analyzers: Vec<Box<dyn FileAnalyzer>>,
}

impl FileAnalyzers {
  fn process_chunk_impl(&mut self, chunk: &[u8]) -> Result<()> {
    for analyzer in self.analyzers {
      analyzer.process_chunk(chunk)?;
    }
  }
}
```

Reply:
Yeah, we could probably refactor this if we start to get more summaries, but considering we are moving summaries to custom-views v2 / XetMeta, likely in the next sprint, it isn't too important to refactor these implementations.